Add options parameter, add `ignoreAutomaticGrants` option

netux opened the pull request:
Fixes #10 and lays the ground work for having more options in the future. Backwards compatible too!

netux committed:
feat: add option to ignore automatic grants

netux committed:
feat: merge `transform` into `userOptions`, add overloads to allow for any parameter combination

This allows for either providing just a transform function, just an options object with an (optional) transform function, or both as separate parameters. All parameters being optional!

gera2ld commented on src/index.ts:
```diff
@@ -5,7 +5,32 @@ import { collectGrants, getMetadata } from './util';
+function userscriptPlugin(
+  transform?: TransformFn,
+  options?: Omit<UserscriptMetaOptions, 'transform'>
+): Plugin;
```
No need to support two parameters because the object options already contains the transform function.

netux replied:
In this case, no it doesn't. Because of the `Omit<..., 'transform'>`.

Or are you saying it the two-parameter overload is unnecessary because users can opt-in to this `ignoreAutomaticGrants` already by just passing an object with a `transform` function in it?

gera2ld replied:
Yes, I mean user can pass everything as an object. There is no need to support separate parameters.

netux committed:
feat: remove 2-parameter overload

You can now either only pass a transform function, an object with a transform function and other options, or nothing at all.

As requested in a code review.
